parquet-column/src/main/java/org/apache/parquet/column/statistics/geometry/GeospatialStatistics.java

```diff
@@ -0,0 +1,260 @@
+     * Build a GeospatialStatistics from the builder.
+     *
+     * @return a new GeospatialStatistics object
+     */
+    public GeospatialStatistics build() {
+      return new GeospatialStatistics(boundingBox, geospatialTypes, edgeAlgorithm);
+    }
+  }
+
+  /**
+   * Create a new GeospatialStatistics builder with the specified CRS.
+   *
+   * @param type the primitive type
+   * @return a new GeospatialStatistics builder
+   */
+  public static GeospatialStatistics.Builder newBuilder(PrimitiveType type) {
+    LogicalTypeAnnotation logicalTypeAnnotation = type.getLogicalTypeAnnotation();
+    if (logicalTypeAnnotation instanceof LogicalTypeAnnotation.GeometryLogicalTypeAnnotation) {
+      return new GeospatialStatistics.Builder();
+    } else if (logicalTypeAnnotation instanceof LogicalTypeAnnotation.GeographyLogicalTypeAnnotation) {
+      EdgeInterpolationAlgorithm edgeAlgorithm =
+          ((LogicalTypeAnnotation.GeographyLogicalTypeAnnotation) logicalTypeAnnotation).getAlgorithm();
+      return new GeospatialStatistics.Builder(edgeAlgorithm);
+    } else {
+      return noopBuilder();
+    }
+  }
+
+  /**
+   * Constructs a GeospatialStatistics object with the specified CRS, bounding box, and geospatial types.
+   *
+   * @param boundingBox the bounding box for the geospatial data, or null if not applicable, note that
+   *    - The bounding box (bbox) is omitted only if there are no X or Y values.
+   *    - The Z and/or M statistics are omitted only if there are no Z and/or M values, respectively.
+   * @param geospatialTypes the geospatial types
+   */
+  public GeospatialStatistics(
+      BoundingBox boundingBox, GeospatialTypes geospatialTypes, EdgeInterpolationAlgorithm edgeAlgorithm) {
+    this.boundingBox = boundingBox;
+    this.geospatialTypes = geospatialTypes;
+    this.edgeAlgorithm = edgeAlgorithm;
+  }
+
+  /**
+   * Constructs a GeospatialStatistics object with the specified CRS.
+   */
+  public GeospatialStatistics() {
+    this(new BoundingBox(), new GeospatialTypes(), null);
+  }
+
+  /**
+   * Constructs a GeospatialStatistics object with the specified CRS and edge interpolation algorithm.
+   *
+   * @param crs the coordinate reference system
+   * @param edgeAlgorithm the edge interpolation algorithm
+   */
+  public GeospatialStatistics(String crs, EdgeInterpolationAlgorithm edgeAlgorithm) {
+    this.boundingBox = new BoundingBox();
+    this.geospatialTypes = new GeospatialTypes();
+    this.edgeAlgorithm = edgeAlgorithm;
+  }
+
+  /** Returns the bounding box. */
+  public BoundingBox getBoundingBox() {
+    return boundingBox;
+  }
+
+  /** Returns the geometry types. */
+  public GeospatialTypes getGeospatialTypes() {
+    return geospatialTypes;
+  }
+
+  /**
+   * @return whether the statistics has valid value.
+   */
+  public boolean isValid() {
+    if (boundingBox == null || geospatialTypes == null) {
+      return false;
+    }
+    return boundingBox.isValid() && geospatialTypes.isValid();
```

Comment:
```suggestion
    return (boundingBox != null && boundingBox.isValid()) ||
      (geospatialTypes != null && geospatialTypes.isValid());
```

We don't need both of them to be alive.